executor/cte_test.go

```diff
@@ -19,64 +19,70 @@ import (
 	"fmt"
 	"math/rand"
 	"sort"
-
-	"github.com/pingcap/check"
+	"testing"
 
 	"github.com/pingcap/tidb/domain"
 	"github.com/pingcap/tidb/kv"
 	"github.com/pingcap/tidb/session"
 	"github.com/pingcap/tidb/sessionctx"
+	"github.com/stretchr/testify/require"
 
 	"github.com/pingcap/failpoint"
 	"github.com/pingcap/tidb/config"
 	"github.com/pingcap/tidb/store/mockstore"
+	"github.com/pingcap/tidb/testkit"
 	"github.com/pingcap/tidb/util/mock"
-	"github.com/pingcap/tidb/util/testkit"
 )
 
-var _ = check.Suite(&CTETestSuite{&baseCTETestSuite{}})
-var _ = check.SerialSuites(&CTESerialTestSuite{&baseCTETestSuite{}})
-
-type baseCTETestSuite struct {
+type CTETestSuite struct {
 	store      kv.Storage
 	dom        *domain.Domain
 	sessionCtx sessionctx.Context
 	session    session.Session
 	ctx        context.Context
+	close      func()
 }
 
-type CTETestSuite struct {
-	*baseCTETestSuite
-}
-
-type CTESerialTestSuite struct {
-	*baseCTETestSuite
-}
+var cteTestSuite *CTETestSuite
 
-func (test *baseCTETestSuite) SetUpSuite(c *check.C) {
+func SetUpSuite(t *testing.T) *CTETestSuite {
 	var err error
+	test := new(CTETestSuite)
+
 	test.store, err = mockstore.NewMockStore()
-	c.Assert(err, check.IsNil)
+	require.NoError(t, err)
 
 	test.dom, err = session.BootstrapSession(test.store)
-	c.Assert(err, check.IsNil)
+	require.NoError(t, err)
 
 	test.sessionCtx = mock.NewContext()
 
 	test.session, err = session.CreateSession4Test(test.store)
-	c.Assert(err, check.IsNil)
+	require.NoError(t, err)
 	test.session.SetConnectionID(0)
 
 	test.ctx = context.Background()
+
+	test.close = func() {
+		test.dom.Close()
+		test.store.Close()
+	}
+
+	return test
 }
 
-func (test *baseCTETestSuite) TearDownSuite(c *check.C) {
-	test.dom.Close()
-	test.store.Close()
+func TestCTESuite(t *testing.T) {
+	cteTestSuite = SetUpSuite(t)
+	defer cteTestSuite.close()
+
+	t.Run("TestBasicCTE", BasicCTE)
```

Comment:
IMO, we don't need the global variable `cteTestSuite` and all subtests can be paralleled, e.g.
```
func TestCTESuite(t *testing.T) {
	t.Run("TestBasicCTE", func(t *testing.T) {
		cteTestSuite := SetUpSuite(t)
		BasicCTE(t, cteTestSuite)
		cteTestSuite.close()
	})
}

func BasicCTE(t *testing.T, cteTestSuite *CTETestSuite) {
	t.Parallel()
	tk := testkit.NewTestKit(t, cteTestSuite.store)
	tk.MustExec("use test")
    ...
}
```

Reply:
In #26712 , it is pointed out that
> In the past, one test suite create one store/domain.
> If after the refactor, each test case create one store/domain, the operation is heavier than before... 
> session.BootstrapSession is not cheap.

So it might be better to create the suite only once, but a global variable is introduced.
If you think it is unnecessary, I will remove it.

Reply:
`So it might be better to create the suite only once` seems reasonable. 
I think remove the global variable is better, we can create `cteTestSuite` once and pass it to subtest as an argument.

Reply:
Fixed. Could you please take a look ?